### PR TITLE
#675 add first draft for tft_explainer

### PR DESCRIPTION
**Pull request by Cattes**
# This is a Draft PR

Fixes #.

### CODE REVIEW

File: darts/explainability/tft_explainer.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            foreground_future_covariates: Optional[
+                Union[TimeSeries, Sequence[TimeSeries]]
+            ] = None,
+            horizons: Optional[Sequence[int]] = None,
+            target_components: Optional[Sequence[str]] = None,
+    ) -> ExplainabilityResult:
+        super().explain(
+            foreground_series, foreground_past_covariates, foreground_future_covariates
+        )
+        if self._explain_results is None:
```
Comment (hrzn) [inline darts/explainability/tft_explainer.py]: I don't think you should cache the explainability results, they should be recomputed when explain() is called (with potentially different series)

File: darts/explainability/tft_explainer.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+)
+from darts.models import TFTModel
+
+
+class TFTExplainer(ForecastingModelExplainer):
+
+    def __init__(
+            self,
+            model: TFTModel,
+            background_series: Optional[Union[TimeSeries, Sequence[TimeSeries]]] = None,
```
Comment (hrzn) [inline darts/explainability/tft_explainer.py]: Do we need to provide these background series? If they don't play a role I'd remove.

File: darts/explainability/tft_explainer.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+            # return the explainer result to be used in other methods
+            expl_res = {
+                "decoder_weights_percentage": decoder_weights_percentage,
+                "decoder_names": decoder_names,
+                "encoder_weights_percentage": encoder_weights_percentage,
+                "encoder_names": encoder_names,
+                "attention_heads": attention_heads,
+            }
+            self._explain_results = ExplainabilityResult({"tft": expl_res})
```
Comment (hrzn) [inline darts/explainability/tft_explainer.py]: The `ExplainabilityResult` should probably contain some `TimeSeries`

**Comment by Cattes** [comment]
This PR is deprecated. The real PR can be found in the darts repository https://github.com/unit8co/darts/pull/1392

**Pull request closed without merging**